cedar-policy-validator/src/err.rs

```diff
@@ -234,6 +242,37 @@ impl From<transitive_closure::TcError<EntityUID>> for SchemaError {
     }
 }
 
+impl SchemaError {
+    /// Given one or more `SchemaError`, collect them into a single `SchemaError`.
+    /// Due to current structures, some errors may have to be dropped in some cases.
+    pub fn join_nonempty(errs: NonEmpty<SchemaError>) -> SchemaError {
+        // if we have any `TypeResolutionError`s, we can report all of those at once (but have to drop the others).
+        // Same for `ActionResolutionError`s.
+        // Any other error, we can just report the first one and have to drop the others.
+        let (type_res_errors, other_errors): (Vec<_>, Vec<_>) =
+            errs.into_iter().partition_map(|e| match e {
+                SchemaError::TypeResolution(e) => Either::Left(e),
+                _ => Either::Right(e),
+            });
+        if let Some(errs) = NonEmpty::from_vec(type_res_errors) {
+            schema_errors::TypeResolutionError::join_nonempty(errs).into()
+        } else {
+            let (action_res_errors, other_errors): (Vec<_>, Vec<_>) =
+                other_errors.into_iter().partition_map(|e| match e {
+                    SchemaError::ActionResolution(e) => Either::Left(e),
+                    _ => Either::Right(e),
+                });
+            if let Some(errs) = NonEmpty::from_vec(action_res_errors) {
+                schema_errors::ActionResolutionError::join_nonempty(errs).into()
+            } else {
+                // PANIC SAFETY: `other_errors` was created by partitioning a `NonEmpty` into what we now know is an empty vector and this, so this cannot be empty
```

Comment:
nit: this `other_errors` was created by partition the shadowed `other_errors` which is a `Vec`, but shadowed `other_errors` wasn't empty by the same logic applied here.

Reply:
Updated this comment